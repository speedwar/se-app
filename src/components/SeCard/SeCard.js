import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { bp, colors } from 'styles'

const SeCard = ({ imgLogo, imgTile, title, provider, href }) => (
  <Card>
    <CardContainer
      href={ href }
      aria-label={`Click to watch ${title}`}
    >
      <CardContent>
        <CardImageResponsive>
          <CardProgramImage
            src={ imgTile }
            alt={ title }
          />
        </CardImageResponsive>
        <CardContentBottom>
          <CardContentImg
            src={ imgLogo }
            alt={ provider}
          />
          <CardContentTitle>
            { title }
          </CardContentTitle>
        </CardContentBottom>
      </CardContent>
    </CardContainer>
  </Card>
)


SeCard.propTypes = {
  imgLogo: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
  ]),
  imgTile: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
  ]),
  title: PropTypes.string.isRequired,
  provider: PropTypes.string.isRequired,
  href: PropTypes.string,
}

SeCard.defaultProps = {
  href: '#',
}

export default SeCard

const Card = styled.div`
  display: inline-block;
  position: relative;
  padding-right: .5em;
  padding-left: .5em;
  margin-bottom: 2.5em;
  width: 100%;

  ${bp.greaterThan('xs')`
    width: 50%;
  `};

  ${bp.greaterThan('sm')`
    width: 33.33%;
  `}

  ${bp.greaterThan('md')`
    width: 25%;
  `}

  ${bp.greaterThan('lg')`
    width: 340px;
  `}
`

const CardContainer = styled.a``

const CardProgramImage = styled.img`
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 0;
`

const CardContent = styled.div`
  position: relative;
`

const CardContentBottom = styled.div`
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.6);
  height: 55px;
  padding: 0 6px;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
`

const CardContentImg = styled.img`
  width: 35px;
  height: 35px;
  margin-right: 6px;
`

const CardContentTitle = styled.p`
  color: ${colors.white};
  font-size: 16px;
  font-weight: 700;
`

const CardImageResponsive = styled.div`
  padding-top: 56.25%;
  height: 0;
  position: relative;
`
