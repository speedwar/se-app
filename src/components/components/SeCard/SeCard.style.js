export const card = `
  position: relative;
  width: 100%;
`

export const cardContent = `
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.6);
  height: 55px;
  padding: 0 6px;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  align-items: center;
`

export const cardContentImg = `
  width: 35px;
  height: 35px;
  margin-right: 6px;
`

export const styles = {
  card,
  cardContent,
  cardContentImg,
}
