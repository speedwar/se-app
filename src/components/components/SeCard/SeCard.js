import React from 'react'
import PropTypes from 'prop-types'
import * as style from './SeCard.style'

const SeCard = ({ imgLogo, imgTile, title, provider, href }) => {
  return (
    <style.Card>
      <style.CardClickable
        href={ href }
        aria-label={`Click to watch ${title}`}
      >
        <style.CardContent>
          <style.CardImageResponsive>
            <style.CardProgramImage
              src={ imgTile }
              alt={ title } 
            />
          </style.CardImageResponsive>
          <style.CardContentBottom>
            <style.CardContentImg
              src={ imgLogo }
              alt={ provider}
            />
            <style.CardContentTitle>
              { title }
            </style.CardContentTitle>
          </style.CardContentBottom>
        </style.CardContent>
      </style.CardClickable>
    </style.Card>
  )
}

SeCard.propTypes = {
  imgLogo: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
  ]),
  imgTile: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
  ]),
  title: PropTypes.string.isRequired,
  provider: PropTypes.string.isRequired,
  href: PropTypes.string,
}

SeCard.defaultProps = {
  href: '#',
}

export default SeCard
