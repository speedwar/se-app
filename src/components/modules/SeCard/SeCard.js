import React, { memo, useContext } from 'react'
import styled from 'styled-components'
import { ThemeContext } from 'components/context'
import { styles } from './SeCard.style'

const SeCard = ({ logo, image, title, href }) => {
  const theme = useContext(ThemeContext)

  const Card = styled.div`
    ${styles.card}
  `

  const CardClickable = styled.a``

  const CardProgramImage = styled.img`
    width: 100%;
  `

  const CardContent = styled.div`
    ${styles.cardContent}
  `

  const CardContentImg = styled.img`
    ${styles.cardContentImg}
  `

  const CardContentTitle = styled.p`
    color: ${theme.white}
    font-size: 16px;
    font-weight: 700;
  `

  return (
    <Card>
      <CardClickable
        href={ href || '#' }
        aria-label={`Navigate to ${title} page`}
      >
        <CardProgramImage
          src={ image }
          alt="Card image"
        />
        <CardContent>
          <CardContentImg
            src={ logo }
            alt="Seven logo"
          />
          <CardContentTitle>
            { title }
          </CardContentTitle>
        </CardContent>
      </CardClickable>
    </Card>
  )
}

export default memo(SeCard)
