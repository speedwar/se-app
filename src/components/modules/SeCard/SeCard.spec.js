import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import SeCard from './SeCard'

describe('<SeCard /> component', () => {
  it('renders component with props', () => {
    const wrapper = shallow(<SeCard />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders component with styles', () => {
    const tree = renderer.create(<SeCard />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('position', expect.stringContaining('relative'))
    expect(tree).toHaveStyleRule('width', '315px')
    expect(tree).not.toHaveStyleRule('margin')
    expect(tree).not.toHaveStyleRule('padding')
  })
})
