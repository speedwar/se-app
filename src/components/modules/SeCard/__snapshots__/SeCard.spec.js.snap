// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeCard /> component renders component with props 1`] = `
<SeCard__Card>
  <SeCard__CardClickable
    aria-label="Navigate to undefined page"
    href="#"
  >
    <SeCard__CardProgramImage
      alt="Card image"
    />
    <SeCard__CardContent>
      <SeCard__CardContentImg
        alt="Seven logo"
      />
      <SeCard__CardContentTitle />
    </SeCard__CardContent>
  </SeCard__CardClickable>
</SeCard__Card>
`;

exports[`<SeCard /> component renders component with styles 1`] = `
.c0 {
  position: relative;
  width: 315px;
  height: 177px;
}

.c1 {
  width: 100%;
}

.c2 {
  background-color: rgb(0,0,0);
  background-color: rgba(0,0,0,0.6);
  height: 55px;
  padding: 0 6px;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 35px;
  height: 35px;
  margin-right: 6px;
}

.c4 {
  color: font-size:16px;
  font-weight: 700;
}

<div
  className="c0"
>
  <a
    aria-label="Navigate to undefined page"
    className=""
    href="#"
  >
    <img
      alt="Card image"
      className="c1"
    />
    <div
      className="c2"
    >
      <img
        alt="Seven logo"
        className="c3"
      />
      <p
        className="c4"
      />
    </div>
  </a>
</div>
`;
